### PR TITLE
Ka/data type rule.w 7275902

### PR DESCRIPTION
**Pull request by kalagarraj**
- Added upper-camelcase-datatype rule to mercury profile
- Added upper-camelcase-datatype rule to slas profile
- Added data type definition test cases to mercury profile test

### CODE REVIEW

**Review by gkhunger-sf — DISMISSED** [review]

File: resources/lint/profiles/mercury.raml [hunk 1]
```diff
@@ -198,3 +199,12 @@ validations:
...
         pattern: ^v[0-9]+$
         minCount: 1
+
+  #datatype rules
+  upper-camelcase-datatype:
+    message: Data type definitions should be UpperCamelCase
+    targetClass: doc.Unit
+    propertyConstraints:
+      doc.declares/shacl.name:
+        pattern: ^[A-Z]+([A-Z]?[a-z0-9]?)*$
```
Comment (bredmond-sf) [inline resources/lint/profiles/mercury.raml:209]:
```suggestion
        pattern: ^[A-Z][A-Za-z0-9]*$
```
This is the simplified version of the same pattern

File: resources/lint/profiles/slas.raml [hunk 2]
```diff
@@ -198,3 +199,11 @@ validations:
...
         pattern: ^v[0-9]+$
         minCount: 1
+
+  #datatype rules
+  upper-camelcase-datatype:
+    message: Data type definitions should be UpperCamelCase
+    targetClass: doc.Unit
+    propertyConstraints:
+      doc.declares/shacl.name:
+        pattern: ^[A-Z]+([A-Z]?[a-z0-9]?)*$
```
Comment (bredmond-sf) [inline resources/lint/profiles/slas.raml:209]: Same as my other comment

File: resources/lint/profiles/mercury.raml (hunk 1 above)
Comment (kalagarraj) [reply resources/lint/profiles/mercury.raml]: Done

File: resources/lint/profiles/slas.raml (hunk 2 above)
Comment (kalagarraj) [reply resources/lint/profiles/slas.raml]: Done

**Review by gkhunger-sf — APPROVED** [review]

**Review by bredmond-sf — APPROVED** [review]

**Pull request merged**